### ISSUE TITLE
libwebp-base 1.2.3 seems broken on macOS (osx-64 and osx-arm64)

### ISSUE THREAD

**Issue by ktlim** [conda-forge/libwebp-base-feedstock#11]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

libwebp-base 1.2.3, which came out a few hours ago, changed build systems from autotools to cmake.  In the process, the macOS dylib that is created went from `@rpath/libwebp.7.dylib (compatibility version 9.0.0, current version 9.3.0)` to `@rpath/libwebp.7.dylib (compatibility version 7.0.0, current version 7.1.4)`, as reported by `otool -L`.

The result is that other libraries cannot successfully link to this one, in particular `libtiff`.  This is actively causing link failures in our code and CI systems.  We believe the build should be pulled.

The [instructions](https://conda-forge.org/docs/maintainer/knowledge_base.html#moving-from-an-autotools-build-to-a-cmake-build) call out in step 2 that the `otool -L` output should be the same.

Credit goes to @erykoff for providing the details above.

### Installed packages

```shell
List of packages in environment: "/Users/ktl/micromamba/envs/webp"
  Name          Version  Build       Channel
─────────────────────────────────────────────────────────
  jpeg          9e       hac89ed1_2  conda-forge/osx-64
  lerc          3.0      he49afe7_0  conda-forge/osx-64
  libcxx        14.0.6   hce7ea42_0  conda-forge/osx-64
  libdeflate    1.12     hac89ed1_0  conda-forge/osx-64
  libtiff       4.4.0    h9847915_1  conda-forge/osx-64
  libwebp-base  1.2.3    hac89ed1_0  conda-forge/osx-64
  libzlib       1.2.12   hfe4f2af_2  conda-forge/osx-64
  lz4-c         1.9.3    he49afe7_1  conda-forge/osx-64
  xz            5.2.5    haf1e3a3_1  conda-forge/osx-64
  zstd          1.5.2    ha9df2e0_2  conda-forge/osx-64
```

### Environment info

```shell
environment : webp (active)
           env location : /Users/ktl/micromamba/envs/webp
      user config files : /Users/ktl/.mambarc
 populated config files : /Users/ktl/.condarc
     micromamba version : 0.17.0
       virtual packages : __unix=0=0
                          __osx=12.3=0
                          __archspec=1=x86_64
       base environment : /Users/ktl/micromamba
               platform : osx-64
```

**Comment by hmaarrfk** [comment]
cc: @conda-forge/libwebp not sure how transfering issues works so pinging again just in case.

**Comment by carterbox** [comment]
Thanks for submitting the request to pull the ABI breaking package.

@conda-forge/help-c-cpp, Any suggestions on how to transition the osx package from the old to new ABI?

Also, it seems that the metadata generated by autotools for the osx releases is incorrect because it is out of sync with the SONAME, but I don't own a Mac, so I can't investigate to see when it became out of sync or why.

https://abi-laboratory.pro/index.php?view=timeline&l=libwebp

**Comment by hmaarrfk** [comment]
https://conda-forge.org/docs/maintainer/knowledge_base.html?highlight=abi#moving-from-an-autotools-build-to-a-cmake-build

and 

https://github.com/conda-forge/conda-forge.github.io/pull/1688

should help you get going.

I think I never had the time to add a step where users should submit a repodata patch adding an upper bound to the latest available package before the breakage

**Comment by carterbox** [comment]
> Hot-fix the repodata of dependencies to include corrected pinnings for the package.

My understanding is that this means patching old already released builds, but there is no explanation on how to do this. How do I do this?

**Comment by erykoff** [comment]
I don't think we can patch the old builds; that ship has sailed, because (say) existing libtiff is built against the older version, and you'd have to rebuild everything.  I think that either we (a) force SOVERSION on macos to be 9 (but this is hard to maintain!); or (b) go back to autotools build; or (c) bump the version (upstream).  Currently the conda forge pinning for libwebp='1' so we'd need to bump to version 2 to keep the semantic versioning from getting confused.  None of these is great, and I'm probably missing a better option.
I'm trying to think of manipulating the global pinning more cleverly but the fact that it's currently at 1 it really expects that all versions reporting as 1.x.x are compatible with already built binaries.

**Comment by carterbox** [comment]
My plan was to migrate the global pinning to 1.2.2, so all packages pick up the new run_export Then once that was complete, migrate the global pinning to >=1.2.3.

I was leaning towards migration instead of reverting because it seems to me that the SONAME should match the dylib "compatibility version" and the CMAKE build system makes it easier to split the outputs between the base and non-base package, BUT also I found this comment from upstream developers that the autotools build is [the "canonical build"](https://bugs.chromium.org/p/webp/issues/detail?id=575#c2), so I should probably stay with that one.

**Comment by hmaarrfk** [comment]
https://conda-forge.org/docs/orga/guidelines.html?highlight=broken#fixing-broken-packages

The "examples" are all in the python file that people add to.

You can see for example how a missing dependency is added
https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/292/files

**Comment by hmaarrfk** [comment]
> I don't think we can patch the old builds;

You can't patch old builds, but you can patch the repo-data which will stop people from downloading incompatible versions. That is, somewhat good enough.

**Comment by h-vetinari** [comment]
I'm surprised you got broken by a _lower_ compatibility version now being advertised...?

In any case, I fear #13 might not have solved this, as the builds still seem to pick up the lower compatibility version even with the autotools build - checkout the versioned symlinks being packaged for [linux](https://github.com/regro/libcfgraph/blob/master/artifacts/libwebp-base/conda-forge/linux-64/libwebp-base-1.2.3-h166bdaf_1.json#L211-L222) & [unix](https://github.com/regro/libcfgraph/blob/0cbc99a825fcf1412994e8eac4e855047f6044b9/artifacts/libwebp-base/conda-forge/osx-64/libwebp-base-1.2.3-hac89ed1_1.json#L208-L215)

**Comment by erykoff** [comment]
To be clear, the _name_ of the file is `libwebp.7.dylib` but the autotools build advertises itself as `9` and that's what `libtiff` expects.  I have no idea how any of this versioning works.

**Comment by h-vetinari** [comment]
> To be clear, the _name_ of the file is `libwebp.7.dylib` but the autotools build advertises itself as `9` and that's what `libtiff` expects. I have no idea how any of this versioning works.

That in and of itself should be considered a bug (and it's IMO quite-to-very likely CMake was doing the correct thing here). The role of the ".7" in the `.so` / `.dylib` is explicitly to document the ABI version, and should match whatever is advertised through other channels (e.g. `otool`)

**Comment by erykoff** [comment]
That sounds reasonable!  And yet that is not how the build has been working with autotools.  🤷

**Comment by h-vetinari** [comment]
After some communication with upstream (in the context of forwarding a patch we came up with for #14), there's now a [bug](https://bugs.chromium.org/p/webp/issues/detail?id=576) for this divergence that's intended to be fixed for 1.3.0 (AFAICT, the plan is to make the CMake output match autotools; meaning we might be able to switch to CMake after that again).

**Comment by h-vetinari** [comment]
Change (for windows libname) [merged](https://chromium.googlesource.com/webm/libwebp/+/9ac25bcb3986f290d55bd65efd85f27ad8827e98%5E%21/) upstream; the cmake builds should now produce the same output as nmake.
